cloudify_agent/worker.py

```diff
@@ -71,6 +69,50 @@ def _print_task(self, ctx, action, status=None):
                 workflow_id=ctx.get('workflow_id'),
                 suffix=suffix))
 
+    @staticmethod
+    def _validate_not_cancelled(handler, ctx):
+        """
+        This method will validate if the current running tasks is cancelled
+        or not
+        :param handler:
+        :param ctx:
+        """
+        # We need also to handle old tasks still in queue and not picked by
+        # the worker so that we can ignore them as the state of the
+        # execution is cancelled and ignore pending tasks picked by the
+        # worker but still not executed. Morever,we need to handle a case when
+        # resume workflow is running while there are some old operations
+        # tasks still in the queue which holds an invalid execution token
+        # which could raise 401 error
+        # Need to use the context associated with the that task
+        with state.current_ctx.push(handler.ctx):
+            try:
+                # Get the status of the current execution so that we can
+                # tell if the current running task can be run or not
+                current_execution = handler.ctx.get_execution(
+                    ctx.get('execution_id')
+                )
+                if current_execution:
+                    logger.info(
+                        'The current status of the execution is {}'
```

Comment:
@mabuaisha this needs to be `{0}` for py26-compat (the `{}` syntax is new in 2.7)

Reply:
Done and the build now is passed